constant/metadata.go

```diff
@@ -72,22 +73,26 @@ func (t Type) MarshalJSON() ([]byte, error) {
 
 // Metadata is used to store connection address
 type Metadata struct {
-	NetWork     NetWork `json:"network"`
-	Type        Type    `json:"type"`
-	SrcIP       net.IP  `json:"sourceIP"`
-	DstIP       net.IP  `json:"destinationIP"`
-	SrcPort     string  `json:"sourcePort"`
-	DstPort     string  `json:"destinationPort"`
-	AddrType    int     `json:"-"`
-	Host        string  `json:"host"`
-	DNSMode     DNSMode `json:"dnsMode"`
-	Process     string  `json:"process"`
-	ProcessPath string  `json:"processPath"`
-	UserAgent   string  `json:"userAgent"`
+	NetWork     NetWork    `json:"network"`
+	Type        Type       `json:"type"`
+	SrcIP       netip.Addr `json:"sourceIP"`
+	DstIP       netip.Addr `json:"destinationIP"`
+	SrcPort     string     `json:"sourcePort"`
+	DstPort     string     `json:"destinationPort"`
+	AddrType    int        `json:"-"`
+	Host        string     `json:"host"`
+	DNSMode     DNSMode    `json:"dnsMode"`
+	Process     string     `json:"process"`
+	ProcessPath string     `json:"processPath"`
+	UserAgent   string     `json:"userAgent"`
 }
 
 func (m *Metadata) RemoteAddress() string {
-	if m.DstIP != nil {
+	return net.JoinHostPort(m.String(), m.DstPort)
+}
+
+func (m *Metadata) RemoteIPAddress() string {
```

Comment:
what's the difference between `func (m *Metadata) RemoteIPAddress() string {` and `func (m *Metadata) Pure() *Metadata {`

Reply:
string vs. *Metatata ?

Reply:
before calls function `DialContext()` the function `Pure()` has been called.
https://github.com/yaling888/clash/blob/d763900b14d7e0a6dc3933e6fceeebb84ff1b136/tunnel/tunnel.go#L323

Reply:
if you disabled Clash dns, maybe do this is better.
```suggestion
func (m *Metadata) Pure() *Metadata {
	- if m.DNSMode == DNSMapping && m.DstIP.IsValid() {
	+ if m.DNSMode != DNSFakeIP && m.DstIP.IsValid() {
	....
```

Reply:
let me have some think about dns mode. The problem I'm having is on Direct, a domian  resolve to multi ip, if I want ssh connect one of the ip, may be resolved to an other one. problem like ssh ->  1.1.1.1:22 -> dns map(xxx.com)->2.2.2.2:22

Reply:
looks like not need RemoteIPAddress any more. I'll revert it.